e2e/cypress/integration/enterprise/guest_accounts/guest_experience_ui_spec.js

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. #. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/**
+ * Note: This test requires Enterprise license to be uploaded
+ */
+import * as TIMEOUTS from '../../../fixtures/timeouts';
+
+describe('Guest Account - Guest User Experience', () => {
+    before(() => {
+        // # Enable Guest Account Settings
+        cy.apiUpdateConfig({
+            GuestAccountsSettings: {
+                Enable: true,
+            },
+        });
+
+        // # Login as a guest user and go to /
+        cy.loginAsNewGuestUser();
+        cy.visit('/');
+    });
+
+    it('MM-18043 Verify Guest User Restrictions', () => {
+        // *Verify Reduced Options in Main Menu
+        cy.get('#sidebarHeaderDropdownButton').should('be.visible').click();
+        const missingMainOptions = ['#invitePeople', '#teamSettings', '#manageMembers', '#createTeam', '#joinTeam', '#integrations', '#systemConsole'];
+        const includeMainOptions = ['#accountSettings', '#viewMembers', '#leaveTeam'];
+        missingMainOptions.forEach((missingOption) => {
+            cy.get(missingOption).should('not.exist');
+        });
+        includeMainOptions.forEach((includeOption) => {
+            cy.get(includeOption).should('be.visible');
+        });
+
+        // *Verify Reduced Options in LHS
+        const missingLHSOptions = ['#createPublicChannel', '#morePublicButton', '#createPrivateChannel'];
+        missingLHSOptions.forEach((missingOption) => {
+            cy.get(missingOption).should('not.exist');
+        });
+
+        // * Verify Guest Badge in Channel Header
+        cy.get('#sidebarItem_town-square').click({force: true});
+        cy.get('#channelHeaderDescription').within(($el) => {
+            cy.wrap($el).find('.has-guest-header').should('be.visible').and('have.text', 'This channel has guests');
+        });
+
+        // * Verify list of Users and Guest Badge in Channel Members List
+        cy.get('#member_popover').click();
+        cy.get('#member-list-popover').should('be.visible').within(($el) => {
+            cy.wrap($el).find('.more-modal__body > div').
+                children().should('have.length', 2);
+            cy.wrap($el).find('.Badge').should('be.visible').and('have.text', 'GUEST');
```

Comment:
There are only 2 users. Can we verify which user has a badge or not?

Reply:
Ok sure. Let me check and do the needful changes

Reply:
Hi @saturninoabril 
Made the necessary changes